src/System.CommandLine.Subsystems/Subsystems/Annotations/ValueAnnotations.cs

```diff
@@ -14,18 +14,17 @@ public static class ValueAnnotations
     /// Default value for an option or argument
     /// </summary>
     /// <remarks>
-    /// Although the type is <see cref="object?"/>, it must actually be the same type as the type
-    /// parameter of the <see cref="CliArgument{T}"/> or <see cref="CliOption{T}"/>.
+    /// Must be actually the same type as the type parameter of
+    /// the <see cref="CliArgument{T}"/> or <see cref="CliOption{T}"/>.
     /// </remarks>
-    public static AnnotationId<object?> DefaultValue { get; } = new(Prefix, nameof(DefaultValue));
+    public static AnnotationId DefaultValue { get; } = new(Prefix, nameof(DefaultValue));
 
     /// <summary>
     /// Default value calculation for an option or argument
     /// </summary>
     /// <remarks>
-    /// Although the type is <see cref="object?"/>, it must actually be a <see cref="Func{TResult}">
-    /// with a type parameter matching the the type parameter type of the <see cref="CliArgument{T}"/>
-    /// or <see cref="CliOption{T}"/>
+    /// Must be return a <see cref="Func{TValue}"> with the same type parameter as
+    /// the <see cref="CliArgument{T}"/> or <see cref="CliOption{T}"/>.
```

Comment:
Would it be better to have a "Please use the extension methods and not call this directly" type comment here? Same on line 17.

Reply:
It's probably worth doing it some point, but we'd have to add it to all the annotation ID definitions, not just `ValueAnnotations`. All the methods that let you actually do anything with an `AnnotationId` already have a remark to that effect, so it'd be more of a discoverability thing than correctness thing.

Reply:
That sounds fine. Let's merge and I'll rebase the next PR :smile: